From: Carlos Garnacho <carlosg@gnome.org>
Date: Thu, 8 Dec 2022 20:01:33 +0100
Subject: backends/native: Initialize keyboard a11y on startup

The MetaSeatImpl is tracking changes on keyboard a11y setting changes,
but missing its initialization on startup.

(cherry picked from commit 4e0ffba5c13b0cbb7501976e878db2ddbb57a86b)

Bug: https://gitlab.gnome.org/GNOME/mutter/-/issues/1858
Forwarded: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2750
Forwarded: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2763
Origin: upstream, 43.3, commit:36549b45fd8f824580668654000897c86d451147
---
 src/backends/native/meta-seat-impl.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/backends/native/meta-seat-impl.c b/src/backends/native/meta-seat-impl.c
index 96a7b22..886cfc4 100644
--- a/src/backends/native/meta-seat-impl.c
+++ b/src/backends/native/meta-seat-impl.c
@@ -2818,6 +2818,8 @@ input_thread (MetaSeatImpl *seat_impl)
 {
   MetaSeatImplPrivate *priv = meta_seat_impl_get_instance_private (seat_impl);
   struct xkb_keymap *xkb_keymap;
+  MetaKbdA11ySettings kbd_a11y_settings;
+  MetaInputDeviceNative *keyboard_native;
 
   g_main_context_push_thread_default (seat_impl->input_context);
 
@@ -2865,6 +2867,12 @@ input_thread (MetaSeatImpl *seat_impl)
   seat_impl->has_tablet_switch = has_tablet_switch (seat_impl);
   update_touch_mode (seat_impl);
 
+  keyboard_native = META_INPUT_DEVICE_NATIVE (seat_impl->core_keyboard);
+  meta_input_settings_get_kbd_a11y_settings (seat_impl->input_settings,
+                                             &kbd_a11y_settings);
+  meta_input_device_native_apply_kbd_a11y_settings_in_impl (keyboard_native,
+                                                            &kbd_a11y_settings);
+
   g_mutex_lock (&seat_impl->init_mutex);
   seat_impl->input_thread_initialized = TRUE;
   g_cond_signal (&seat_impl->init_cond);
